### PR TITLE
fix(liveFilterUpdate): Handle conversion of `liveFilterUpdate` from boolean to string.

### PR DESCRIPTION
**Pull request by Camwyn**
Ensure that any values from the database that are not `automatic` or `manual` get converted.

Related: https://github.com/moderntribe/events-pro/pull/1505

[FBAR-83]

[FBAR-83]: https://moderntribe.atlassian.net/browse/FBAR-83

### CODE REVIEW

File: src/Tribe/Views/V2/Hooks.php [hunk 1]
```diff
@@ -90,6 +90,9 @@ protected function add_filters() {
 		add_filter( 'tribe_get_event_after', [ $this, 'filter_events_properties' ] );
 		add_filter( 'tribe_template_file', [ $this, 'filter_template_file' ], 10, 3 );
 		add_filter( 'tribe_get_option', [ $this, 'filter_get_stylesheet_option' ], 10, 2 );
+		add_filter( 'option_liveFiltersUpdate', [ $this, 'filter_live_filters_option_value' ], 10, 2 );
+		add_filter( 'tribe_get_option', [ $this, 'filter_live_filters_option_value' ], 10, 2 );
+		add_filter( 'tribe_field_value', [ $this, 'filter_live_filters_option_value'], 10, 2 );
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/Hooks.php]: * :no_entry_sign: Missing space before array closer.
* :no_entry_sign: Missing space before array closer.

<sub>hash: 85fbcb8f7e9ac8bfc81aaff0c7f717fb</sub>

File: src/Tribe/Views/V2/Hooks.php [hunk 2]
```diff
@@ -635,4 +638,45 @@ public function filter_get_stylesheet_option( $value, $key ) {
...
+	 * Filter the liveFiltersUpdate option to do some switching for V2.
+	 * Note: this triggers on option_liveFiltersUpdate, tribe_get_option, AND tribe_field_value. We
+	 * don't have to add/remove filters because we don't need to get the value - it's already provided.
+	 *
+	 * @since TBD
+	 *
+	 * @param string $value      The option value.
+	 * @param string $optionName The option key.
+	 *
+	 * @return void
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/Hooks.php]: :no_entry_sign: Function return type is void, but function contains return statement

<sub>hash: 97503e4dd9c1ceb0cc9d8820e7284ed5</sub>

File: src/Tribe/Views/V2/Hooks.php [hunk 3]
```diff
@@ -635,4 +638,45 @@ public function filter_get_stylesheet_option( $value, $key ) {
...
+	 */
+	public function filter_live_filters_option_value( $value, $optionName ) {
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/Hooks.php]: :no_entry_sign: Variable "$optionName" is not in valid snake_case format, try "$option_name"

<sub>hash: c7c60f29ab14a286478ff834809eb18b</sub>

File: src/Tribe/Views/V2/Hooks.php [hunk 4]
```diff
@@ -635,4 +638,45 @@ public function filter_get_stylesheet_option( $value, $key ) {
...
+		if ( 'liveFiltersUpdate' !== $optionName ) {
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/Hooks.php]: :no_entry_sign: Variable "$optionName" is not in valid snake_case format, try "$option_name"

<sub>hash: 4b193c499fd73a78ebf8ecb01f26d7eb</sub>

File: src/admin-views/tribe-options-general.php [hunk 5]
```diff
@@ -270,6 +269,12 @@
 		return $fields;
 	} );
 } else {
+	if ( tribe_is_truthy( tribe_get_option( 'tribeDisableTribeBar', false ) ) ) {
+		$filter_activation[ 'attributes' ] = [ 'disabled' => 'disabled' ];
```
Comment (tr1b0t) [inline src/admin-views/tribe-options-general.php]: :no_entry_sign: Array keys must NOT be surrounded by spaces if they only contain a string or an integer.

<sub>hash: 465f0c6dbb318be35165039e1e3e964c</sub>

File: src/admin-views/tribe-options-general.php [hunk 6]
```diff
@@ -270,6 +269,12 @@
...
+		$filter_activation[ 'class' ]      = 'tribe-fieldset-disabled';
```
Comment (tr1b0t) [inline src/admin-views/tribe-options-general.php]: :no_entry_sign: Array keys must NOT be surrounded by spaces if they only contain a string or an integer.

<sub>hash: d1d49867bbb152439c9735d1def02131</sub>

File: src/admin-views/tribe-options-general.php [hunk 7]
```diff
@@ -270,6 +269,12 @@
...
+		$filter_activation[ 'tooltip' ]    = esc_html__( 'This option is disabled when "Disable the Event Search Bar" is checked on the Display settings tab.', 'the-events-calendar' );
```
Comment (tr1b0t) [inline src/admin-views/tribe-options-general.php]: :no_entry_sign: Array keys must NOT be surrounded by spaces if they only contain a string or an integer.

<sub>hash: 1bf9f9784c3c313f2ae6edd86db5c12d</sub>

File: src/Tribe/Views/V2/Hooks.php [hunk 8]
```diff
@@ -635,4 +638,45 @@ public function filter_get_stylesheet_option( $value, $key ) {
...
+
+	/**
+	 * Filter the liveFiltersUpdate option to do some switching for V2.
+	 * Note: this triggers on option_liveFiltersUpdate, tribe_get_option, AND tribe_field_value. We
+	 * don't have to add/remove filters because we don't need to get the value - it's already provided.
+	 *
+	 * @since TBD
+	 *
+	 * @param string $value      The option value.
+	 * @param string $key The option key.
```
Comment (lucatume) [inline src/Tribe/Views/V2/Hooks.php]: Align the two doc lines.

File: src/Tribe/Views/V2/Hooks.php [hunk 9]
```diff
@@ -635,4 +638,45 @@ public function filter_get_stylesheet_option( $value, $key ) {
...
+	 *
+	 * @return string
```
Comment (lucatume) [inline src/Tribe/Views/V2/Hooks.php]: Add return tag doc entry.

File: src/Tribe/Views/V2/Hooks.php [hunk 10]
```diff
@@ -635,4 +638,45 @@ public function filter_get_stylesheet_option( $value, $key ) {
...
+	}
+
+	/**
+	 * Converts old (boolean) values to the new string values.
+	 *
+	 * @since TBD
+	 *
+	 * @param mixed $value The value to maybe convert.
+	 *
+	 * @return string
```
Comment (lucatume) [inline src/Tribe/Views/V2/Hooks.php]: Add return tag doc entry.

File: src/Tribe/Views/V2/Hooks.php [hunk 11]
```diff
@@ -635,4 +638,53 @@ public function filter_get_stylesheet_option( $value, $key ) {
...
+
+		return $this->live_filters_maybe_convert( $value );
+	}
+
+	/**
+	 * Converts old (boolean) values to the new string values.
+	 *
+	 * @since TBD
+	 *
+	 * @param  mixed  $value The value to maybe convert.
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/Hooks.php]: :no_entry_sign: Expected 1 spaces after parameter type; 2 found

<sub>hash: 536e2096667a6bd58ea1fbc4df7266b8</sub>

**Pull request merged**